sweep/txgenerator_test.go

```diff
@@ -0,0 +1,89 @@
+package sweep
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/input"
+)
+
+// weightTests is a list of all standard witness types and their expected
+// witness size for one input.
+var weightTests = []struct {
+	witnessType    input.WitnessType
+	expectedWeight int64
+}{
+	{
+		witnessType:    input.CommitmentTimeLock,
+		expectedWeight: 486,
+	},
+	{
+		witnessType:    input.CommitmentNoDelay,
+		expectedWeight: 439,
+	},
+	{
+		witnessType:    input.CommitmentRevoke,
+		expectedWeight: 486,
+	},
+	{
+		witnessType:    input.HtlcOfferedRevoke,
+		expectedWeight: 573,
+	},
+	{
+		witnessType:    input.HtlcAcceptedRevoke,
+		expectedWeight: 579,
+	},
+	{
+		witnessType:    input.HtlcOfferedTimeoutSecondLevel,
+		expectedWeight: 486,
+	},
+	{
+		witnessType:    input.HtlcAcceptedSuccessSecondLevel,
+		expectedWeight: 486,
+	},
+	{
+		witnessType:    input.HtlcOfferedRemoteTimeout,
+		expectedWeight: 546,
+	},
+	{
+		witnessType:    input.HtlcAcceptedRemoteSuccess,
+		expectedWeight: 647,
+	},
+	{
+		witnessType:    input.HtlcSecondLevelRevoke,
+		expectedWeight: 486,
+	},
+	{
+		witnessType:    input.WitnessKeyHash,
+		expectedWeight: 439,
+	},
+	{
+		witnessType:    input.NestedWitnessKeyHash,
+		expectedWeight: 579,
+	},
+	{
+		witnessType:    input.CommitSpendNoDelayTweakless,
+		expectedWeight: 439,
+	},
+}
+
+// TestWeightEstimate tests that the estimated weights of the most commonly used
+// sweep witness types are correct.
+func TestWeightEstimate(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range weightTests {
+		_, weight, _, _ := getWeightEstimate([]input.Input{
+			input.NewBaseInput(
+				&wire.OutPoint{}, test.witnessType,
+				&input.SignDescriptor{}, 0,
+			),
+		})
+		if weight != test.expectedWeight {
+			t.Fatalf("unexpected weight for witness type %s. "+
+				"expected %d but got %d",
+				test.witnessType.String(), test.expectedWeight,
+				weight)
+		}
+	}
+}
```

Comment:
What is the purpose of this test? It seems like it is just testing the constants in the code. A different setup is needed to catch the bug that was introduced in the previous revision of this pr. It would need to have multiple components added to the same estimator.

Reply:
Correct, this test is only useful for more than one input. I fixed it and again verified it to be correct against master too.

Reply:
Ok. Only I don't think it is necessary to test all these witness types. What more is it than asserting on constants? Having a single `getWeightEstimate` test to assert that we only use one weight estimator internally seems like enough to me.

Reply:
A simple testcase with mulitple inputs that asserts final weight and CSV/CLTV count is maybe sufficient?

Reply:
Yes, that makes sense. Testing every single type doesn't add much value. So I picked a few that use CSV and CLTV and moved the commit with the test to the beginning. That way we can be assert that the estimation isn't changed by the refactor.

Reply:
Looking good